src/perf_counters.cc

```diff
@@ -57,9 +57,20 @@ size_t PerfCounterValues::Read(const std::vector<int>& leaders) {
 
 const bool PerfCounters::kSupported = true;
 
-bool PerfCounters::Initialize() { return pfm_initialize() == PFM_SUCCESS; }
+// Initializes libpfm only on the first call.  Returns whether that single
+// initialization was successful
+static bool InitLibPfmOnce() {
```

Comment:
I could also delete `InitLibPfmOnce` and put this directly in `Initialize`.

I thought it was clearer to have the other calls use this name, though.

Reply:
i think having it directly in `Initialize` and having the other methods call `Initialize` is clearer.

Reply:
i think that's fine.  the callsites are blindly trying to initialize and if it already is, then it's a no-op.

Reply:
Updated.